reco_utils/dataset/movielens.py

```diff
@@ -407,18 +407,10 @@ def _load_datafile(size, local_cache_path):
         with z.open(DATA_FORMAT[size].item_path) as zf, open(item_datapath, 'wb') as f:
             shutil.copyfileobj(zf, f)
 
-    _clean_up(local_cache_path)
+    remove_filepath(local_cache_path)
```

Comment:
a cleaner option would be to leverage temporary directories:
```
from tempfile import TemporaryDirectory

if local_cache_path is not None:
  tmp_dir = TemporaryDirectory()
  path = tmp_dir.name
  filename = 'data.zip'
  atexit.register(tmp.cleanup)
else:
  path, filename = os.path.split(os.path.realpath(local_cache_path))
```

Reply:
changes based on @loomlike PR